prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusRemoteWriteHttpRequestBuilder.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.connector.prometheus.sink;
+
+import org.apache.flink.util.Preconditions;
+
+import org.apache.hc.client5.http.async.methods.SimpleHttpRequest;
+import org.apache.hc.client5.http.async.methods.SimpleRequestBuilder;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpHeaders;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/** Builds the POST request to the Remote-Write endpoint for a given binary payload. */
+public class PrometheusRemoteWriteHttpRequestBuilder {
+
+    private static final ContentType CONTENT_TYPE = ContentType.create("application/x-protobuf");
+
+    public static final String DEFAULT_USER_AGENT = "Flink-Prometheus";
+    private final String prometheusRemoteWriteUrl;
+    private final PrometheusRequestSigner requestSigner;
+
+    private final Map<String, String> fixedHeaders;
+
+    public PrometheusRemoteWriteHttpRequestBuilder(
+            String prometheusRemoteWriteUrl,
+            PrometheusRequestSigner requestSigner,
+            String httpUserAgent) {
+        Preconditions.checkNotNull(httpUserAgent, "User-Agent not specified");
+
+        this.prometheusRemoteWriteUrl = prometheusRemoteWriteUrl;
+        this.requestSigner = requestSigner;
+        this.fixedHeaders =
+                Map.of(
+                        HttpHeaders.CONTENT_ENCODING,
+                        "snappy",
+                        "X-Prometheus-Remote-Write-Version",
+                        "0.1.0",
+                        HttpHeaders.USER_AGENT,
+                        httpUserAgent);
+    }
+
+    public SimpleHttpRequest buildHttpRequest(byte[] httpRequestBody) {
+        Map<String, String> headers = new HashMap<>(fixedHeaders);
+        if (requestSigner != null) {
+            requestSigner.addSignatureHeaders(headers, httpRequestBody);
+        }
+
+        var builder =
```

Comment:
Default target java version for connectors is 8, is java 11 features necessary?

Reply:
Is it still the case for 1.18?
Anyhow, I removed java 11 features